addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/ClientScriptBasedAuthenticationMethodType.java

```diff
@@ -429,228 +448,45 @@ private static Map<String, String> wrapKeys(Map<String, String> kvPairs) {
-                if (adaptOldValues && dynamicFieldsPanel != null) {
-                    oldValues = dynamicFieldsPanel.getFieldValues();
-                    LOGGER.debug("Trying to adapt old values: {}", oldValues);
+                Field scriptsComboBoxField =
+                        ScriptBasedAuthenticationMethodOptionsPanel.class.getDeclaredField(
+                                "scriptsComboBox");
+                scriptsComboBoxField.setAccessible(true);
+                JXComboBox scriptsCb = (JXComboBox) scriptsComboBoxField.get(this);
+                DefaultComboBoxModel<ScriptWrapper> model =
+                        (DefaultComboBoxModel<ScriptWrapper>) scriptsCb.getModel();
+                for (int i = 0; i < model.getSize(); i++) {
+                    if (!model.getElementAt(i).getEngineName().contains("Zest")) {
+                        model.removeElementAt(i);
+                        i--;
+                    }
                 }
-
-                this.dynamicFieldsPanel = new DynamicFieldsPanel(requiredParams, optionalParams);
-                this.loadedScript = scriptW;
-                if (adaptOldValues && oldValues != null) {
-                    this.dynamicFieldsPanel.bindFieldValues(oldValues);
-                }
-
-                this.dynamicContentPanel.removeAll();
-                this.dynamicContentPanel.add(dynamicFieldsPanel, BorderLayout.CENTER);
-                this.dynamicContentPanel.revalidate();
-
-            } catch (Exception e) {
-                getExtensionScript().handleScriptException(scriptW, e);
-                LOGGER.error("Error while calling authentication script", e);
-                warnAndResetPanel(
-                        Constant.messages.getString(
-                                "authentication.method.script.dialog.error.text.loading",
-                                ExceptionUtils.getRootCauseMessage(e)));
+            } catch (Exception ignore) {
             }
         }
 
-        private void warnAndResetPanel(String errorMessage) {
-            JOptionPane.showMessageDialog(
-                    this,
-                    errorMessage,
-                    Constant.messages.getString("authentication.method.script.dialog.error.title"),
-                    JOptionPane.ERROR_MESSAGE);
-            this.loadedScript = null;
-            this.scriptsComboBox.setSelectedItem(null);
-            this.dynamicFieldsPanel = null;
-            this.dynamicContentPanel.removeAll();
-            this.dynamicContentPanel.add(new JLabel(LABEL_NOT_LOADED), BorderLayout.CENTER);
-            this.dynamicContentPanel.revalidate();
+        // @Override
+        protected List<ScriptWrapper> getAuthenticationScripts() {
+            // TODO return super.getAuthenticationScripts().stream()
+            return getExtensionScript().getScripts(SCRIPT_TYPE_AUTH).stream()
+                    .filter(sc -> sc.getEngineName().contains("Zest"))
```

Comment:
Do we not have "Zest" as a constant somewhere? It's used here a few times.

Reply:
A few is two and once the TODO is done it will be just one. (There's no constant.)